backend/src/hmc-api/fetcher/fetch.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable */
+// we need to do this because https://github.com/DefinitelyTyped/DefinitelyTyped/issues/60924
+// once this is resolved we can also remove the dependency on undici
+import * as process from "process";
+
+declare global {
+    export const {
+        fetch,
+        FormData,
+        Headers,
+        Request,
+        Response,
+    }: typeof import("undici");
+}
+export {};
+/* eslint-enable */
+import { createLogger } from "../../logger";
+import { endpoints, endpointAuthorization } from "./endpoints";
+import * as APIv4 from "hyperschedule-shared/api/v4";
+import type { Endpoint } from "./types";
+import { computeParams, saveStatic } from "./utils";
+
+const logger = createLogger("hmc.fetcher");
+
+async function doFetch(endpoint: Endpoint, term: APIv4.TermIdentifier) {
+    const params = computeParams(term);
+    let link: string;
+    if (endpoint.param === null) link = endpoint.link;
+    else
+        link = `${endpoint.link}?${new URLSearchParams({
+            [endpoint.param.toUpperCase()]: params[endpoint.param],
+        }).toString()}`;
```

Comment:
purely aesthetic personal preference but any chance I could invite you to do this with a `const` ternary?

```typescript
const link = endpoint.param === null ? endpoint.link : ...
```

Reply:
(i like `const` over `let` wherever possible, even though practically there isn't much of a difference)